### PR TITLE
Patch from `osx` to build ICL on mac

### PR DESCRIPTION
**Pull request by aleneum**
This makes CMAKE more flexible and allows to also build `dylib` for OSX builts.
Additionally, there is a script which can be used to test homebrew builds locally first.

### CODE REVIEW

**Review by scires — APPROVED** [review]

**Review by rhaschke — CHANGES_REQUESTED** [review]
Looks fine so far. I would integrate `scripts/icl-homebrew-test.sh` into `.travis.yaml`.

**Comment by celbrech** [comment]
looks good to me, well done!

File: .travis.yml [hunk 1]
```diff
@@ -58,4 +58,4 @@ matrix:
     # - <<: *EXTENDED
     #   compiler: clang
     - os: osx
-      script: sh scripts/icl-homebrew-test
+      script: sh scripts/icl-homebrew-test.sh
```
Comment (rhaschke) [inline .travis.yml]: Actually, I meant to not simply _call_ the script from Travis, but to _get rid of_ the script and use the [Travis' way to install osx dependencies](https://docs.travis-ci.com/user/reference/osx) and finally build as usual.

**Review by rhaschke — CHANGES_REQUESTED** [review]
.travis.yml should be finally extended with osx stuff.

**Comment by aleneum** [comment]
closing this for now until osx job runs stable

**Pull request closed without merging**